students/Net_Michael/lesson3/mailroom.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+# Part I
+
+import sys  
+
+def sort_key(donor_db2):
+        return donor_db2[0].split(" ")[1]
+
+# update the list of donors and donation
+def add_donor():
+    new_donor = input("Name of donor?").title()
+    if new_donor not in donor_name:
+        donor_name.append(new_donor)  
+        new_donation1 = input("Amount of donation by {}?".format(new_donor))
+        idx = len(donor_name)-1
+        new_donation1 = float(new_donation1)
+        donation[idx].append(new_donation1)
+    elif new_donor in donor_name:
+        idx = donor_name.index(new_donor)
+        new_donation1 = float(input("Amount of donation?".format(new_donor)))
+        donation[idx].append(new_donation1)
+    print(f"{new_donor} has made a new donation of {new_donation1}")
+    return(new_donor, new_donation1)
+
+
+# compile donors and donation information
+def donors_info():
+    
+    mm = len(donor_name)
+    donor_db2 = [''] * mm
+    
+    for j in range(mm):
+        donor_db2[j] = (donor_name[j], donation[j])
+        
+    donor_db2 = sorted(donor_db2, key=sort_key)
+    return(donor_db2)
+
+# create tabular report 
+
+def create_report():
+    mm = len(donor_name)
+    data_db2 = []*mm
+
+    print("Name\t\t\tTotal Donation\t\tNum Gifts\tAverage Gift")
+
+    for j in range(mm):
+        total_donation = sum(donation[j])
+        num_gifts = len(donation[j])
+        avg_donation = total_donation/num_gifts
+        
+        print('{:23}'.format(donor_name[j]),
+              '${:>10,.2f}'.format(total_donation),
+              '{:>15}'.format(num_gifts),
+              '{:10}'.format(""), 
+              '${:>10,.2f}'.format(avg_donation))
+
+          
+def exit_program():
+    return("Done")  
+
+
+# In[2]:
+
+
+
+# Part I
+# list of donors and history of the amounts they have donated.
+
+donor_name = ["William Gates, III", 
+              "Jeff Bezos", 
+              "Paul Allen", 
+              "Mark Zuckerberg"]
+
+donation = ( [653772.32, 10020.17, 58796.00],
+            [877.33],
+            [663.23, 43.87, 1.32], 
+            [1663.23, 4300.87, 10432.0])
+#part I
+# This structure should be populated at first with at least five donors,
+#with between 1 and 3 donations each.
+#You can store that data structure in the global namespace.
+
+donor_list1 = dict(zip(donor_name, donation))
+donor_list1
```

Comment:
for this lesson you should use data structure learned so far, in this case it should be a `list`


`add_donor` function above is not dealing with the dictionary in an efficient way - since this should be a list my suggestion will be updating donors as a list.